controlplane/internal/controllers/scale.go

```diff
@@ -270,7 +277,7 @@ func (r *RKE2ControlPlaneReconciler) cloneConfigsAndGenerateMachine(ctx context.
 
 	// Only proceed to generating the Machine if we haven't encountered an error
 	if len(errs) == 0 {
-		if err := r.generateMachine(ctx, rcp, cluster, infraRef, bootstrapRef, failureDomain); err != nil {
+		if err := r.generateMachine(ctx, rcp, cluster, infraRef, bootstrapRef); err != nil {
```

Comment:
Same comment regarding needing to use / be aware of failure domains.

Reply:
Re-introduced failure domains using a revert commit and some fixes.